### DISCUSSION TITLE
School Timetable Schedule

### DISCUSSION THREAD

**Discussion by christmex** [Laravel-Backpack/community-forum#414] [Show and Tell]
Hi guys, I create this project for the school timetable, but I am still working on it, a few updates will be coming

here's the repo
https://github.com/christmex/timetable-schedule

I will update how to use it and configure this project soon, thx guys, let me know if I did something wrong in my code, I hope I can learn more to use this great tool "backpackforlaravel".

<img width="540" alt="image" src="https://user-images.githubusercontent.com/34405837/221526920-89652471-a177-4179-848a-bee517f7ccb8.png">

**Comment by tabacitu** [comment]
Hey @christmex ,

Looking good! No time for a full code review, but I had a quick look and some things stood out to me:
- I see in [ScheduleCrudController](https://github.com/christmex/timetable-schedule/blob/main/app/Http/Controllers/Admin/ScheduleCrudController.php#L297) you're overriding the `update()` methods; that's ok if absolutely needed, but nowadays I prefer if possible to [use Model Events inside CrudControllers](https://backpackforlaravel.com/docs/5.x/release-notes#use-model-events-inside-crudcontrollers); that way I don't override anything, I just do stuff before/after updating; so if Backpack updates the `update()` method, I get that update too; depends on use case of course, sometimes overriding is exactly what you need;
- same as the above with `store()`
- in `UserCrudController` I wouldn't show the password inside the List operation; you can see all the changes I'd do to UserCrudController here - https://github.com/Laravel-Backpack/CRUD/blob/main/src/resources/views/base/inc/getting_started.blade.php but you have to "read HTML" 😅

Hope it helps.
Cheers!

**Reply by christmex** [reply]
Thank you so much for the feedback, that's really helpful. I almost overwrite store() and update() functions in all of my backpackforlaravel projects.
